src/_provider/singleAbstract.ts

```diff
@@ -94,6 +101,60 @@ export abstract class SingleAbstract {
     return this._getStatus();
   }
 
+  abstract _setStartDate(startDate: definitions.startFinishDate): void;
+
+  public setStartDate(startDate: definitions.startFinishDate): SingleAbstract {
```

Comment:
startDate could be null too right?

Reply:
Resolved by adjusting `definitions.startFinishDate` to be `string | null` and removing the other `| null`